File: feedforward_prop.m
function [p, time] = feedforward_prop(X, Theta1, Theta2, Theta3)

tic;

m = size(X, 1);

p = zeros(1, 850);

X = [ones(m, 1) X];

a_2 = sigmoid(X * Theta1);

a_2 = [ones(size(a_2), 1) a_2];

a_3 = sigmoid(a_2 * Theta2);

a_3 = [ones(size(a_3), 1) a_3];

p = sigmoid(a_3 * Theta3);

time = toc;

end

File: feedforward_prop.py
import numpy as np
import time


def sigmoid(z):
    return 1.0 / (1.0 + np.exp(-z))


def feedforward_prop(X, Theta1, Theta2, Theta3):
    X = np.hstack((1, X))

    a_2 = sigmoid(X @ Theta1)

    a_2 = np.hstack((1, a_2))

    a_3 = sigmoid(a_2 @ Theta2)

    a_3 = np.hstack((1, a_3))

    return sigmoid(a_3 @ Theta3)
